api/src/main/java/org/apache/iceberg/RewriteFiles.java

```diff
@@ -34,13 +34,72 @@
  * will throw a {@link ValidationException}.
  */
 public interface RewriteFiles extends SnapshotUpdate<RewriteFiles> {
+  /**
+   * Delete a data file whose content was rewritten.
+   *
+   * @param dataFile a rewritten data file
+   * @return this for method chaining
+   */
+  default RewriteFiles deleteFile(DataFile dataFile) {
+    throw new UnsupportedOperationException(
+        this.getClass().getName() + " does not implement deleteFile");
+  }
+
+  /**
+   * Delete a delete file whose content was rewritten.
+   *
+   * @param deleteFile a rewritten delete file
+   * @return this for method chaining
+   */
+  default RewriteFiles deleteFile(DeleteFile deleteFile) {
+    throw new UnsupportedOperationException(
+        this.getClass().getName() + " does not implement deleteFile");
+  }
+
+  /**
+   * Add a new data file.
+   *
+   * @param dataFile a new data file
+   * @return this for method chaining
+   */
+  default RewriteFiles addFile(DataFile dataFile) {
+    throw new UnsupportedOperationException(
+        this.getClass().getName() + " does not implement addFile");
+  }
+
+  /**
+   * Add a new delete file.
+   *
+   * @param deleteFile a new delete file
+   * @return this for method chaining
+   */
+  default RewriteFiles addFile(DeleteFile deleteFile) {
```

Comment:
Do we want to specify that adding the same delete file/data file twice will have no effect? We required set's before

Reply:
That's a good point, I'll add. That said, neither `DataFile` nor `DeleteFile` implementations provide proper equality at the moment so `Set` was kind of useless.

Reply:
True, Sounds like we can drop it, although may want to fix that in the future since we don't actually support multiple instances of the same file within a table.